src/app/components/core/filter/filter-list/filter-list.component.ts

```diff
@@ -94,22 +96,20 @@ export class FilterListComponent {
   }
 
   hasFilter(filter: string) {
-    const filterName = this.type !== 'control-pipeline' ? this.service + '_' + filter : `ctrl_${this.service}_${filter}`;
-    return this.filterConfiguration.has(filterName);
+    return this.filterConfiguration.has(this.getFilterName(filter));
   }
 
   getFilter(filter: string) {
-    const filterName = this.type !== 'control-pipeline' ? this.service + '_' + filter : `ctrl_${this.service}_${filter}`;
-    return this.filterConfiguration.get(filterName);
+    return this.filterConfiguration.get(this.getFilterName(filter));
+  }
+
+  getFilterName(filter: string) {
+    const filterName = this.type !== 'control-pipeline' ? this.service + '_' + filter : ((this.newfilter?.filter === filter) ? filter : `ctrl_${this.service}_${filter}`);
```

Comment:
This is too nested, please use if else

Reply:
use  string interpolation for` this.service + '_' + filter` as well?

What are this.type values?